bin/node/testing/src/bench.rs

```diff
@@ -219,6 +211,93 @@ impl CloneableSpawn for TaskExecutor {
 	}
 }
 
+/// Iterator for block content.
+pub struct BlockContentIterator<'a> {
+	iteration: usize,
+	content: BlockContent,
+	runtime_version: sc_executor::RuntimeVersion,
+	genesis_hash: node_primitives::Hash,
+	keyring: &'a BenchKeyring,
+}
+
+impl<'a> BlockContentIterator<'a> {
+	fn new(content: BlockContent, keyring: &'a BenchKeyring, client: &Client) -> Self {
+		let runtime_version = client.runtime_version_at(&BlockId::number(0))
+			.expect("There should be runtime version at 0");
+
+		let genesis_hash = client.block_hash(Zero::zero())
+			.expect("Database error?")
+			.expect("Genesis block always exists; qed")
+			.into();
+
+		BlockContentIterator {
+			iteration: 0,
+			content,
+			keyring,
+			runtime_version,
+			genesis_hash,
+		}
+	}
+}
+
+impl<'a> Iterator for BlockContentIterator<'a> {
+	type Item = OpaqueExtrinsic;
+
+	fn next(&mut self) -> Option<Self::Item> {
+		if self.content.size.map(|size| size <= self.iteration).unwrap_or(false) {
+			return None;
+		}
+
+		let sender = self.keyring.at(self.iteration);
+		let receiver = get_account_id_from_seed::<sr25519::Public>(
+			&format!("random-user//{}", self.iteration)
+		);
+
+		let signed = self.keyring.sign(
+			CheckedExtrinsic {
+				signed: Some((sender, signed_extra(0, node_runtime::ExistentialDeposit::get() + 1))),
+				function: match self.content.block_type {
+					BlockType::RandomTransfersKeepAlive => {
+						Call::Balances(
+							BalancesCall::transfer_keep_alive(
+								pallet_indices::address::Address::Id(receiver),
+								node_runtime::ExistentialDeposit::get() + 1,
+							)
+						)
+					},
+					BlockType::RandomTransfersReaping => {
+						Call::Balances(
+							BalancesCall::transfer(
+								pallet_indices::address::Address::Id(receiver),
+								// Transfer so that ending balance would be 1 less than existential deposit
+								// so that we kill the sender account.
+								100*DOLLARS - (node_runtime::ExistentialDeposit::get() - 1),
```

Comment:
It was still not clear to me that killing transfer actually worked as expected. I think I remember the import benchmark for killing was faster than the one for keep alive...